pkg/operators/ebpf/attach.go

```diff
@@ -145,6 +153,88 @@ func (i *ebpfInstance) attachProgram(gadgetCtx operators.GadgetContext, p *ebpf.
 		return link.AttachLSM(link.LSMOptions{
 			Program: prog,
 		})
+	case ebpf.PerfEvent:
+		perfType := uint32(unix.PERF_TYPE_SOFTWARE)
+		perfConfig := uint64(unix.PERF_COUNT_SW_CPU_CLOCK)
+		perfSampleType := uint64(unix.PERF_SAMPLE_RAW)
+		frequency := uint64(0)
+		name, ok := strings.CutPrefix(p.SectionName, perfEventPrefix)
+		if !ok {
+			return nil, fmt.Errorf("perf_event programs require a name")
+		}
+
+		if tmp := i.config.GetString("programs." + name + ".perf.type"); tmp != "" {
+			switch tmp {
+			case "software":
+				perfType = unix.PERF_TYPE_SOFTWARE
+			default:
+				return nil, fmt.Errorf("unsupported perf.type %q", tmp)
+			}
+		} else {
+			return nil, fmt.Errorf("perf.type not specified for program %q", name)
+		}
+		if tmp := i.config.GetString("programs." + name + ".perf.config"); tmp != "" {
+			switch tmp {
+			case "count_sw_cpu_clock":
+				perfConfig = unix.PERF_COUNT_SW_CPU_CLOCK
+			default:
+				return nil, fmt.Errorf("unsupported perf.config %q", tmp)
+			}
+		} else {
+			return nil, fmt.Errorf("perf.config not specified for program %q", name)
+		}
+		if tmp := i.config.GetString("programs." + name + ".perf.sampleType"); tmp != "" {
+			switch tmp {
+			case "sample_raw":
+				perfSampleType = unix.PERF_SAMPLE_RAW
+			default:
+				return nil, fmt.Errorf("unsupported perf.sampleType %q", tmp)
+			}
+		} else {
+			return nil, fmt.Errorf("perf.sampleType not specified for program %q", name)
+		}
+		if tmpFrequency := i.config.GetString("programs." + name + ".sampler.frequency"); tmpFrequency != "" {
+			var err error
+			frequency, err = strconv.ParseUint(tmpFrequency, 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("parsing frequency %q for program %q: %w", tmpFrequency, name, err)
+			}
+		} else {
+			return nil, fmt.Errorf("sampler.frequency not specified for program %q", name)
+		}
+		if frequency == 0 {
+			return nil, fmt.Errorf("sampler.frequency is zero for program %q", name)
+		}
```

Comment:
What's the plan regarding this? To remove all the configuration options and leave hardcoded the case we know that works or to keep it configurable?

Reply:
The plan is to have the gadget author specify __all__ parameters as needed (not relying on defaults that could change later on) in order to keep forward compatibility (e.g.: if we choose to add other options later on, existing gadget (hopefullly) won't need an update).

Reply:
IMO it is not necessary to force gadget authors to specify all parameters for forward compatibility.
For example: ".perf.type=software" could be hardcoded. And if we later want to make it configurable, we could have ".perf.type" default to "software". So older gadgets without this configuration would still work.

That would allow to simplify the gadget author job, and those lines could be deleted:
```
programs:
  profiler:
    perf:
      type: software
      config: count_sw_cpu_clock
      sampleType: sample_raw
    sampler:
      frequency: 49
```

Reply:
I also don't have a super strong opinion on this.

About the setting "frequency=49", I wonder if this could be a parameter on the ebpf operator (chosen at runtime) instead of a configuration option in the gadget (chosen at build-time).